lib/phoenix/channel.ex

```diff
@@ -279,16 +293,19 @@ defmodule Phoenix.Channel do
               {:shutdown, :left | :closed} |
               term
 
-  defmacro __using__(_) do
+  defmacro __using__(opts \\ []) do
     quote do
       @behaviour unquote(__MODULE__)
       @on_definition unquote(__MODULE__)
       @before_compile unquote(__MODULE__)
       @phoenix_intercepts []
+      @log unquote(opts[:log] || [join: :info, handle_in: :info])
```

Comment:
`bind_quoted: opts` and the value should be accessed at runtime. We should also do `Keyword.get(opts, :log, [...])` otherwise `log: false` would never work?